build/kernel-link.x

```diff
@@ -70,12 +69,14 @@ SECTIONS
 	KEEP(*(.image_header));
   } > FLASH
 
-  PROVIDE(_stext = ADDR(.vector_table) + SIZEOF(.vector_table) + SIZEOF(.header));
+  PROVIDE(_stext = ADDR(.header) + SIZEOF(.header));
```

Comment:
I'm not sure about changing this definition. This now sets `_stext` relative to the header as opposed to the vector table. It's the same technically but spelling it out explicitly seems clearer.

Reply:
The text will follow the header, the original expression could become wrong if the header's alignment were changed (which would make it more than just a sizeof-offset from the previous section). This expression of it is more robust.

Reply:
Do we want it to be robust or do we want it to fail so we know the alignment has changed? If we can get it to fail I'd prefer that over finding something with the header changed.

Reply:
I believe if I rolled back to what you had, it would fail if the alignment were changed, though I don't totally understand why that's good. I'm willing to do that with a large comment explaining why it's built that way when it looks wrong/ increases the delta.

Reply:
Change rolled back and comment added.

Reply:
My real concern is something silently succeeding. If the alignment has changed without us intending it I don't want other things that may be making that assumption to break. Maybe nothing else cares about the alignment but I'm deeply distrustful of linker scripts at this point.